modules/netaddress.go

```diff
@@ -15,6 +20,9 @@ func (na NetAddress) Host() string {
 	if err != nil {
 		return ""
 	}
+	if na.IsValid() != nil {
+		return ""
+	}
```

Comment:
hmm, not sure about these. `Host` and `Port` are just convenience wrappers more than rigorous methods. It's sort of implicit that the netaddress is valid.